hazelcast/src/test/java/com/hazelcast/replicatedmap/ReplicatedMapMergePolicyTest.java

```diff
@@ -77,30 +69,38 @@ public void killAllHazelcastInstances() {
     @Parameterized.Parameter
     public ReplicatedMapMergePolicyTestCase testCase;
 
+    private TestHazelcastInstanceFactory factory;
```

Comment:
The termination can't be handled by `HazelcastTestSupport` if you don't use its factory and methods. You are creating your own factory in the test, those instances will not be terminated.

Remove
```java
private TestHazelcastInstanceFactory factory;

   @Before
    public void init() {
        factory = createHazelcastInstanceFactory(2);
    }
```
and use
```java
        final HazelcastInstance h1 = createHazelcastInstance(config);
        final HazelcastInstance h2 = createHazelcastInstance(config);
```
to create the instances. That should solve the issue (and needs less code).